doc/BRANCHING-AND-RELEASE.adoc

```diff
@@ -0,0 +1,126 @@
+= Branching Policy and Release Process
+:toc: left
+:reproducible:
+
+== Branching Policy
+
+This repo uses two development branches: `main` and `next-node`.
+`main` tracks the latest Cardano node version.
+`next-node` tracks either Cardano node master, or a Cardano node release branch.
+
+Using two development branches lets us not only publish new Plutus Tools versions for the current node version, but also quickly publish a new version compatible with a newly-published node version.
+
+The following outlines the sequence of events from one node release, version `nX`, to the next node release, version `nY`.
+Suppose the current version of Plutus Tools is `tX`, compatible with node `nX`.
+Let `PV` denote the current major protocol version, and `ERA` the current ledger era.
+
+There are different kinds of node releases: a node release can be a non-hard fork release, an intra-era hard fork release (i.e., `PV` will be incremented, but not `ERA`), or a next-era hard fork release.
+In the following, assume `nY` to be a next-era hard fork release, which is the most complex case.
+
+- `main` depends on node `nX`, and `next-node` tracks node master.
+* `next-node` should regularly updates its node commit. Ideally we want `next-node` to depend on the tip of node master, but we are constrained by cardano-wallet, which is a PAB dependency and depends on the node.
```

Comment:
```suggestion
* `next-node` should regularly update its node commit. Ideally we want `next-node` to depend on the tip of node master, but we are constrained by cardano-wallet, which is a PAB dependency and depends on the node.
```

Reply:
Do we actually want this, or do we want to track node RCs?

Reply:
I think that in the spirit of integrating with the node as soon as possible, we should track `master` on `next-node`.